From 32521675ded7cabda7e5619708afd0181780ad3e Mon Sep 17 00:00:00 2001
From: Martin Liska <mliska@suse.cz>
Date: Thu, 4 Feb 2021 11:18:05 +0100
Subject: [PATCH] Glibc 2.33 fix.

The library does not declare __fxstatat64 and friends in a header file.
Thus the following warning appears:

```
libsystem.c:333:1: error: no previous prototype for '__xstat' [-Werror=missing-prototypes]
  333 | __xstat (int version, const char *path, struct stat *buf)
      | ^~~~~~~
libsystem.c:342:1: error: no previous prototype for '__xstat64' [-Werror=missing-prototypes]
  342 | __xstat64 (int version, const char *path, struct stat64 *buf)
      | ^~~~~~~~~
libsystem.c:351:1: error: no previous prototype for '__fxstatat' [-Werror=missing-prototypes]
  351 | __fxstatat(int ver, int dirfd, const char *pathname, struct stat *buf, int flags)
      | ^~~~~~~~~~
libsystem.c:360:1: error: no previous prototype for '__fxstatat64' [-Werror=missing-prototypes]
  360 | __fxstatat64(int ver, int dirfd, const char *pathname, struct stat64 *buf, int flags)
      | ^~~~~~~~~~~~
```

Fixed #167.
---
 tests/src/libsystem.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/src/libsystem.c b/tests/src/libsystem.c
index b47b4523..aed3b8cf 100644
--- a/tests/src/libsystem.c
+++ b/tests/src/libsystem.c
@@ -329,6 +329,10 @@ stat64 (const char *path, struct stat64 *buf)
     return _stat64 (new_path, buf);
 }
 
+// glibc 2.33 and newer does not declare these functions in a header file
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-prototypes"
+
 int
 __xstat (int version, const char *path, struct stat *buf)
 {
@@ -365,6 +369,8 @@ __fxstatat64(int ver, int dirfd, const char *pathname, struct stat64 *buf, int f
     return ___fxstatat64 (ver, dirfd, new_path, buf, flags);
 }
 
+#pragma GCC diagnostic pop
+
 DIR *
 opendir (const char *name)
 {
-- 
2.30.0

